### ISSUE TITLE
Warns of unused imports inside of test mod

### ISSUE THREAD

**Issue by N9199** [rust-lang/rust-clippy#8406]
### Summary

I was writing tests for project and found that clippy said that I had unused imports in the module, I trusted clippy removed them and then clippy complained with and error that some structs and methods were undeclared, telling me to import them, the same which I had just removed.

### Reproducer

I tried this code:

```rust

#[derive(PartialEq, Debug)]
pub struct TestStruct {}

impl TestStruct {
    pub fn new() -> Self {
        Self {}
    }
}

trait TestTrait {
    fn do_test(&self);
}

impl TestTrait for TestStruct {
    fn do_test(&self) {
        todo!()
    }
}

mod test {
    use crate::TestStruct;
    use crate::TestTrait;

    #[test]
    fn test() {
        let test_var1 = TestStruct::new();
        let test_var2 = TestStruct::new();
        test_var1.do_test();
        assert_eq!(test_var1, test_var2);
    }
}
```

I saw this happen:

```
warning: unused import: `crate::TestStruct`
  --> src/main.rs:27:9
   |
27 |     use crate::TestStruct;
   |         ^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(unused_imports)]` on by default

warning: unused import: `crate::TestTrait`
  --> src/main.rs:28:9
   |
28 |     use crate::TestTrait;
   |         ^^^^^^^^^^^^^^^^

```

I expected to see this happen:
No warning

### Version

```text
rustc 1.58.1 (db9d1b20b 2022-01-20)
binary: rustc
commit-hash: db9d1b20bba1968c1ec1fc49616d4742c1725b4b
commit-date: 2022-01-20
host: x86_64-unknown-linux-gnu
release: 1.58.1
LLVM version: 13.0.0
```

**Comment by llogiq** [comment]
`unused_imports` is a rustc lint, not a clippy lint (or it would be `clippy::unused_imports`). Also you can solve the problem by putting `#[cfg(test)]` before your `mod test {`.

**Comment by N9199** [comment]
Oh, sorry, missread that it was clippy, but adding #[cfg(test)] doesn't solve it (the code where I found the false positive has it).
